src/main.rs

```diff
@@ -75,53 +63,35 @@ struct Newer {
 
 impl Newer {
     fn new(args: Vec<String>) -> Result<Newer, Error> {
-        enum State {
-            Output,
-            Input,
-            Command,
-        }
+        let mut args = args.iter();
+
+        let newest = args
+            .by_ref()
+            .take_while(|&a| a != ":")
```

Comment:
This seems like it's no longer finding the newest outputs, only the newest inputs, or am I mistaken?

Reply:
In short:
```rust
newest = inputs.map { |i| find_newest(i) }.max().unwrap_or(SystemTime::UNIX_EPOCH)
newers = outputs.map { |o| find_newest(o) }
newer = if newers.is_empty() {
    true
} else {
    newers.any(|n| n > newest)
};
```

That is:
1. `newest` is the timestamp of the newest input.
2. `newers` are the timestamps of the outputs.
3. `newer` is true if any of the `newers` are more recent then the `newest` or if there are no `newers`.

Did I misunderstand the original logic?